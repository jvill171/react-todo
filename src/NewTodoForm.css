.NewTodoForm > * {
    margin: 0 5px;
    height: 35px;
}

.NewTodoForm > button{
    background-color: rgba(255, 255, 255, 0.507);
    border-radius: 10px;
    padding: 0 15px;
    font-size: 1rem;
}