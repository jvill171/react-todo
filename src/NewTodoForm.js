import React, { useState } from "react";
import "./NewTodoForm.css"

const NewTodoForm = ({addTodo})=>{
    const INITIAL_STATE = {
        "todoContent": ""
    }
    const [formData, setFormData] = useState(INITIAL_STATE)
    const handleChange = (e)=>{
        const {name, value} = e.target
        setFormData(data => ({...data, [name]: value}))
    }
    const handleSubmit = (e)=>{
        e.preventDefault();
        addTodo({...formData})
        setFormData(INITIAL_STATE)
        
    }
    return (
        <form className="NewTodoForm" onSubmit={handleSubmit}>
            <label htmlFor="todoContent">Todo:</label>
            <input
                type="text"
                name="todoContent"
                id="todoContent"
                placeholder="New Todo..."
                value={formData.todoContent}
                onChange={handleChange}
            />
            <button>Add New Todo</button>
        </form>
    )
}

export default NewTodoForm;