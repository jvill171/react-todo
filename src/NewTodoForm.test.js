import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

// Smoke test
test('renders without crashing', () => {
  render(<NewTodoForm />);
});

// Snapshot test
it("matches snapshot", function() {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});