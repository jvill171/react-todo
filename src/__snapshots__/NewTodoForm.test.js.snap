// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <label
      for="todoContent"
    >
      Todo:
    </label>
    <input
      id="todoContent"
      name="todoContent"
      placeholder="New Todo..."
      type="text"
      value=""
    />
    <button>
      Add New Todo
    </button>
  </form>
</DocumentFragment>
`;
